### PR TITLE
Updated formatting guidelines

### PR DESCRIPTION
**Pull request by TheKodeToad**
Aims to reflect how we are doing things now - or perhaps should be 😆 
Might be a little excessive

One thing I don't know about is whether we want to do PascalCase or SCREAMING_SNAKE_CASE enums. I just stuck with the latter as it is traditionally what is prefered.

### CODE REVIEW

**Review by Trial97 — APPROVED** [review]

**Review by TayouVR — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,16 +2,54 @@
...
+All files are formatted with `clang-format` using the configuration in `.clang-format`. Ensure it is run on changed files before committing!
 
-In general, in order of importance:
+We have no tool for enforcing names but please follow the following conventions for C++:
+- Class and type names should be formatted as `PascalCase`: `MyClass`.
+- Private, non-`static const` class data members should be formatted as `camelCase` prefixed with `m_`: `m_myCounter`.
+- Public, non-`static const` should be the same but without the prefix: `dateOfBirth`. 
+- Class function members should be formatted as `camelCase` without a prefix: `incrementCounter`.
+- Non-`const` global variables should be formatted as `camelCase` without a prefix: `globalData`. And, of course, should be avoided where possible.
+- `const` global variables, `static const` class data members, macros, and enum constants should be formatted as `SCREAMING_SNAKE_CASE`: `LIGHT_GRAY`.
```
Comment (Ryex) [inline CONTRIBUTING.md]:
```suggestion
- Private `static` non-const class data members should be formatted as `camelCase` prefixed with `s_`: `s_staticDataMember`.
- `const` global variables,  `static const` class data members, macros, and enum constants should be formatted as `SCREAMING_SNAKE_CASE`: `LIGHT_GRAY`.
```

**Review by Ryex — CHANGES_REQUESTED** [review]

**Review by Ryex — APPROVED** [review]

**Review by getchoo — APPROVED** [review]

**Review by timoreo22 — APPROVED** [review]

**Comment by Ryex** [comment]
Btw, [these conventions could be enforced via clang-tidy](https://clang.llvm.org/extra/clang-tidy/checks/readability/identifier-naming.html#readability-identifier-naming)
we already have the `.clang-tidy` file

**Comment by TheKodeToad** [comment]
If we automatically change everything it will create a load of conflicts though

**Comment by Trial97** [comment]
I'm fine with rebasing all my PRs, we also did this with clang-format
Either in one go or each time we work on a PR we would have a commit dedicated to the rename.

**Pull request merged**